src/nemos/basis.py

```diff
@@ -493,7 +566,7 @@ def _check_transform_input(
             # make sure array is at least 1d (so that we succeed when only
             # passed a scalar)
             xi = tuple(np.atleast_1d(np.asarray(x, dtype=float)) for x in xi)
-        except TypeError:
+        except (TypeError, ValueError):
```

Comment:
this was needed to surface exception obtained passing samples like `np.array(["a",  "b"])`